class-instant-articles-amp-markup.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Facebook Instant Articles for WP.
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package default
+ * @since 4.0
+ */
+
+use Facebook\InstantArticles\AMP\AMPArticle;
+use Facebook\InstantArticles\Validators\Type;
+
+class Instant_Articles_Amp_Markup {
+
+	const SETTING_AMP_MARKUP = 'amp_markup'; /// Setting to check if AMP Markup generation is enabled
+	const SETTING_STYLE      = 'amp_stylesheet'; /// Setting that stores the JSON stylesheet
+	const SETTING_DL_MEDIA   = 'amp_download_media'; /// Enable or disable media download option
+	const QUERY_ARG          = 'amp_markup'; /// Query argument that will trigger the AMP markup generation
+
+	// To memoize the settings
+	static $settings = null;
+
+	/**
+	 * Gets the settings
+	 *
+	 * @return array The settings, check Instant_Articles_Option::get_option_decoded()
+	 * @since 4.0
+	 */
+	static function get_settings() {
+		if (self::$settings === null)
```

Comment:
💅  some code standard nits: 

- always use the full `if` notation with brackets;
- add padding to parenthesis `( self::$settings === null )`